cmd/create/operatorroles/common_utils.go

```diff
@@ -60,3 +64,49 @@ func validateIngressOperatorPolicyOverride(r *rosa.Runtime, policyArn string, sh
 
 	return nil
 }
+
+func getHcpSharedVpcPolicy(r *rosa.Runtime, roleArn string, roleName string,
+	operator *cmv1.STSOperator, path string, defaultPolicyVersion string) (string, error) {
+	policyDetails := "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": {\n    \"Effect\": \"Allow\",\n    " +
+		"\"Action\": \"sts:AssumeRole\",\n    \"Resource\": \"%{shared_vpc_role_arn}\"\n  }\n}\n"
+	policyDetails = aws.InterpolatePolicyDocument(r.Creator.Partition, policyDetails, map[string]string{
+		"shared_vpc_role_arn": roleArn,
+	})
+	policy := aws.GetOperatorPolicyARN(r.Creator.Partition, r.Creator.AccountID,
+		aws.SharedVpcAssumeRolePrefix+"-"+roleName, operator.Namespace(), operator.Name(), path)
+
+	var err error
+	policyArn, err := r.AWSClient.EnsurePolicy(policy, policyDetails, defaultPolicyVersion,
+		map[string]string{tags.RedHatManaged: helper.True}, path)
+	if err != nil {
+		return policyArn, err
+	}
+	return policyArn, nil
+}
+
+func getHcpSharedVpcPolicyDetails(r *rosa.Runtime, roleArn string, roleName string, iamTags map[string]string,
+	path string) (string, string) {
+	policyDetails := ` \
```

Comment:
@hunterkepley Can we extract this to a `const`? It looks the same as the definition of the `policyDetails` variable on line 70?

Reply:
I can extract to a const, but sadly it is not the same as the var on `L70`. Using the var on `L70` does not work with the AWS CLI's `--policy-document` body, and will not create the policy. It must be a byte string with no `\n`'s

Technically I could create a file like managed policies, but it is much cleaner and simpler for the user like this